src/components/NurseryWithdrawals/PlantingProgressMap.tsx

```diff
@@ -92,29 +106,63 @@ export default function PlantingProgressMap({ plantingSiteId }: PlantingProgress
       );
       setRequestId(request.requestId);
       setFocusEntities([]);
+      setConfirmDeleteCb(() => () => null);
       setDispatching(true);
     },
     [dispatch]
   );
 
+  const updatePlantingComplete = useCallback(
+    (id: number, val: boolean) => {
+      if (!selectedZoneHasStats) {
+        completeUpdate(id, val)();
+      } else {
+        setConfirmDeleteCb(() => completeUpdate(id, val));
+        setStatsWarningDialogOpen(true);
+      }
+    },
+    [selectedZoneHasStats, completeUpdate]
+  );
+
+  const onOpenMapDialog = useCallback(
+    (id: number) => {
+      const selectedZone = plantingSite?.plantingZones?.find((zone) =>
+        zone.plantingSubzones.map((sz) => sz.id).includes(id)
+      );
+      setZoneIdSelected(selectedZone?.id ?? -1);
```

Comment:
do we need the -1 fallback?

Reply:
It could be undefined instead of -1 maybe